storage/store.go

```diff
@@ -929,6 +929,11 @@ func (s *Store) Descriptor(nodeDesc *NodeDescriptor) (*StoreDescriptor, error) {
 // method, args & reply into a Raft Cmd struct and executes the
 // command using the fetched range.
 func (s *Store) ExecuteCmd(method string, args proto.Request, reply proto.Response) error {
+	select {
```

Comment:
The problem is that in this test we don't have one giant scope; we have multiple stores that are stopped and started independently (#705). But you're right that this is the wrong place for this check. This points to deeper problems with this test setup: we have multiple stores all sharing a client and a LocalSender. This is wrong on several levels. 
* When we destroy and recreate the store the LocalSender still has a reference to the old version. 
* The LocalSender assumes that each range only appears once on the "local" node, so it routes incoming commands randomly (map order); raft forwarding is the only reason it works at all (the raft transport is separate from the sender).

These tests will need to be updated to use a DistSender (and then I think we need some calls to StartTask in the server side). The evidently untested assumptions of the sender implementations are troubling (in LocalSender, that each RaftID only appears on one store; in DistSender that StoreIDs are globally unique). I also wonder if we can refactor the raft transport to actually use the Sender so there is no way for them to diverge.